enterprise/api/v1/views.py

```diff
@@ -308,23 +327,57 @@ def courses(self, request, enterprise_customer, pk=None):  # pylint: disable=inv
 
         Only courses with active course runs are returned. A course run is considered active if it is currently
         open for enrollment, or will open in the future.
-        ---
-        serializer: serializers.CourseSerializerExcludingClosedRuns
         """
-        catalog_api = CourseCatalogApiClient(request.user)
-        courses = catalog_api.get_paginated_catalog_courses(pk, request.GET)
-
-        # if API returned an empty response, that means pagination has ended.
-        # An empty response can also means that there was a problem fetching data from catalog API.
-        if not courses:
-            logger.error(
-                "Unable to fetch API response for catalog courses from endpoint '%s'.",
-                request.get_full_path(),
-            )
-            raise NotFound("The resource you are looking for does not exist.")
-
-        serializer = serializers.EnterpriseCatalogCoursesReadOnlySerializer(courses)
+        catalogs_api = CatalogsApiClient(request.user)
+        courses = catalogs_api.get_paginated_catalog_courses(pk, request.GET)
+        self.ensure_data_exists(request, courses)
+        serializer = serializers.EnterpriseCoursesReadOnlySerializer(courses)
 
         # Add enterprise related context for the courses.
-        serializer.update_enterprise_courses(enterprise_customer, pk)
+        serializer.update_enterprise_courses(enterprise_customer, catalog_id=pk)
+        return get_paginated_response(serializer.data, request)
+
+
+class EnterpriseProgramsViewSet(EnterpriseWrapperApiViewSet):
```

Comment:
We don't want to make any/all programs in the discovery service available via the Enterprise API. This is where the `EnterpriseCatalog` model comes into play.